pkg/plugins/policies/meshhttproute/xds/configurer.go

```diff
@@ -125,6 +122,46 @@ func (c RoutesConfigurer) routeMethodMatch(envoyMatch *envoy_route.RouteMatch, m
 	)
 }
 
+func routeQueryParamsMatch(envoyMatch *envoy_route.RouteMatch, matches []api.QueryParamsMatch) {
+	// We ignore multiple matchers for the same name, though this is also
+	// validated
+	matchedNames := map[string]struct{}{}
+
+	for _, match := range matches {
+		if _, ok := matchedNames[match.Name]; ok {
+			continue
+		}
+		matchedNames[match.Name] = struct{}{}
+
+		var matcher envoy_type_matcher.StringMatcher
+		switch match.Type {
+		case api.ExactQueryMatch:
+			matcher = envoy_type_matcher.StringMatcher{
+				MatchPattern: &envoy_type_matcher.StringMatcher_Exact{
+					Exact: match.Value,
+				},
+			}
+		case api.RegularExpressionQueryMatch:
+			matcher = envoy_type_matcher.StringMatcher{
+				MatchPattern: &envoy_type_matcher.StringMatcher_SafeRegex{
+					SafeRegex: regexMatcher(match.Value),
+				},
+			}
+		default:
+			panic("impossible")
```

Comment:
nitpick: do we always panic on these things? Let's say someone makes a bug in the future in validation this and it crash loops in prod for someone - not sure it's the right approach :(

Reply:
Yeah it's a good question. If someone adds a new enum variant and it gets released, no matter what we do here, that new variant/functionality won't work. And in order to stop the error/panic, the user need only remove the new variant.

So either way, we must keep the implications of adding a new variant in mind during review. 

At least with a panic, the code can stay clean (no useless `error` return value) and the fact that the functionality is broken is immediately obvious to users and thus to us.